python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1455,6 +1471,15 @@ def mexists(
         -------
         existence : `dict` of [`DatasetRef`, `bool`]
             Mapping from dataset to boolean indicating existence.
+
+        Notes
+        -----
+        To minimize potentially costly remote existence checks, the local
+        cache is checked as a proxy for existence. If a file for this
+        `DatasetRef` does exist no check is done for the actual URI. This
+        could result in possibly unexpected behavior if the dataset itself
+        has been removed from the datastore by another process whilst it is
+        still in the cache.
```

Comment:
Should the check in local cache be combined with the registry check for DatasetId existence, or is it still too expensive?

Reply:
The cache is only checked if the datastore registry already thinks it might exist (or we are in execution butler mode where there might not be a registry entry anyhow).